### DISCUSSION TITLE
Class UART methods in ESP32

### DISCUSSION THREAD

**Discussion by TomLinz** [micropython/micropython#10749] [ESP32] [answered]
With PyBoard, I use MicroPython class UART and its methods UART.any(), UART.readinto()
and UART.write(). I want to port my code to the ESP32-PICO-D4 module, but wonder if 
these three MicroPython methods are implemented in ESP32?
Examples I have found mention only methods UART.write() and UART.read().

**Comment by robert-hh** [comment, marked as answer]
IYou can list all implemented elements of the UART class by:
```
from machine import UART
print(dir(UART))
```

**Reply by TomLinz** [reply]
Thanks Robert, this was a nice answer! Seems the methods I need are implemented. And I learned a valuable new trick :-)

**Reply by TomLinz** [reply]
Further question related to ESP32 UART.init() method. Tried to read back settings of the UART in the same manner. However:
>>> dir(UART.init)
['__class__']

>>> dir(UART.init())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: function missing 1 required positional arguments

>>> dir(UART.init(baudrate))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'baudrate' isn't defined

Is there any other way to read back default settings of UART.init() method, such as txbuf size?

**Comment by shariltumin** [comment]
You can do this (as shown by the following example):

```
MPY: soft reboot
MicroPython v1.20.0-487-g00930b213-dirty on 2023-09-24; ESP32S3-WROOM-1 N16R8 UART (KAKI5) with ESP32-S3
>>> from machine import UART
>>> urt1 = UART(1)
>>> print(urt1)
UART(1, baudrate=115211, bits=8, parity=None, stop=1, tx=10, rx=9, rts=-1, cts=-1, txbuf=256, rxbuf=256, timeout=0, timeout_char=0)
```

**Reply by TomLinz** [reply]
Thanks alot, this works fine for me!
